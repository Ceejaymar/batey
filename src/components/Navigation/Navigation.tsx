import { Link, NavLink as NLink } from 'react-router-dom';
import styled from 'styled-components';
import { ShoppingCart } from '@phosphor-icons/react';

const Nav = styled.nav`
  display: flex;
  align-items: center;
  gap: 2rem;
`;

const Brand = styled.div`
  font-family: 'EB Garamond';
  font-size: 2rem;
  font-weight: 600;
`;

const NavList = styled.ul`
  display: flex;
  gap: 1rem;
  margin-left: auto;
  list-style: none;
`;

const NavLink = styled(NLink)`
  text-decoration: none;
  font-size: 1.3rem;
  color: black;
  letter-spacing: -0.04rem;
`;

const Navigation = () => {
  return (
    <Nav data-testid="navigation">
      <Brand>Batey</Brand>
      <NavList>
        <li>
          <NavLink to="/">Home</NavLink>
        </li>
        <li>
          <NavLink to="about">About</NavLink>
        </li>
        <li>
          <NavLink to="shop">Shop</NavLink>
        </li>
        <li>
          <NavLink to="contact">Contact</NavLink>
        </li>
      </NavList>
      <div>
        <Link to="cart">
          <ShoppingCart
            aria-label="View shopping cart"
            size={28}
            weight="thin"
            color="black"
          />
        </Link>
      </div>
    </Nav>
  );
};

export default Navigation;
