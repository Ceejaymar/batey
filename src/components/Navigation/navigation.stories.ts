import type { Meta, StoryObj } from '@storybook/react';
import Navigation from './Navigation';

const meta = {
  title: 'Navigation',
  component: Navigation,
} satisfies Meta<typeof Navigation>;

export default meta;

type Story = StoryObj<typeof Navigation>;

export const Default: Story = {};
