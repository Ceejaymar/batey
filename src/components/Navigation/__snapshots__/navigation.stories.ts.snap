// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Default smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Home",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "About",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Shop",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Contact",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "View shopping cart",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
  ],
  "name": "navigation--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`components/Navigation Default smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Home",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "About",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Shop",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Contact",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "View shopping cart",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
  ],
  "name": "components-navigation--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
